lib/adapters/sequelize.js

```diff
@@ -41,24 +41,124 @@ function translateOptions(spell, options) {
   if (offset) spell.$offset(offset);
 }
 
+const setScopeToSpell = (scope) => (spell) => {
+  if (scope.where) {
+    spell.$where(scope.where);
+  }
+  if (scope.order) {
+    spell.$order(scope.order);
+  }
+  if (scope.limit) {
+    spell.$limit(scope.limit);
+  }
+};
+
+const mergeScope = (scopes) => {
+  let merged = {};
+  for (const scope of scopes) {
+    if (scope.where) {
+      merged.where = Object.assign({}, merged.where, scope.where);
+    }
+    if (scope.order) {
+      merged.order = Object.assign({}, merged.order, scope.order);
+    }
+    if (scope.limit) {
+      merged.limit = scope.limit;
+    }
+  }
+  return merged;
+};
+
+/**
+ * parse scope
+ * @param {Object} scope
+ * @param {Spell} spell
+ * @returns
+ */
+function parseScope(scope) {
+  if (!scope) return null;
+  if (isPlainObject(scope)) {
+    return setScopeToSpell(scope);
+  }
+  // scope function
+  if (typeof scope === 'function') {
+    scope.__isFunc = true;
+    return scope;
+  }
+  if (Array.isArray(scope)) {
+    // array should not contain function
+    return setScopeToSpell(mergeScope(scope));
+  }
+  return scope;
+}
+
 // https://sequelize.org/master/class/lib/model.js~Model.html
 // https://sequelize.org/master/manual/model-querying-finders.html
 module.exports = Bone => {
   return class Spine extends Bone {
 
+    // store all configured scopes
+    // protect
+    static _scopes = {}
+    // scope
+    static _scope = null;
```

Comment:
does `_scope` serve as the default scope?

Reply:
`_scopes` store all named scopes, it can be inherited to subclass. `_scope` is kind of ”class static private variable“,  we can modify it through `setScope`, and it will be used by `Spell`